### DISCUSSION TITLE
Need to apply different gradients to stroke for <Area type="line" />

### DISCUSSION THREAD

**Discussion by cernanb** [recharts/recharts#4925] [Q&A] [answered]
I need to be able to create something like the following. 

![Screenshot 2024-08-26 at 8 10 57 AM](https://github.com/user-attachments/assets/52306cdd-bb1e-4796-ae84-0fb17948d417)

Imagine there are more data points and I potentially need strokes from point to point to have a different gradients. Can someone take this simple AreaChart code taken from [the docs](https://recharts.org/en-US/api/AreaChart) 

```javascript
const data = [
  {
    "name": "Page A",
    "uv": 4000,
    "amt": 2400
  },
  {
    "name": "Page B",
    "uv": 3000,
    "amt": 2210
  },
  {
    "name": "Page C",
    "uv": 2000,
    "amt": 2290
  }
]

<AreaChart width={730} height={250} data={data}
  margin={{ top: 10, right: 30, left: 0, bottom: 0 }}>
  <defs>
    <linearGradient id="colorUv" x1="0" y1="0" x2="0" y2="1">
      <stop offset="5%" stopColor="#8884d8" stopOpacity={0.8}/>
      <stop offset="95%" stopColor="#8884d8" stopOpacity={0}/>
    </linearGradient>
    <linearGradient id="colorPv" x1="0" y1="0" x2="0" y2="1">
      <stop offset="5%" stopColor="#82ca9d" stopOpacity={0.8}/>
      <stop offset="95%" stopColor="#82ca9d" stopOpacity={0}/>
    </linearGradient>
  </defs>
  <XAxis dataKey="name" />
  <YAxis />
  <CartesianGrid strokeDasharray="3 3" />
  <Tooltip />
  <Area type="line" dataKey="uv" stroke="#8884d8" fillOpacity={1} fill="url(#colorUv)" strokeWidth={4} />
</AreaChart>
```
and show how you would make the strokes from point 1 to point 2 and from point 2 to point 3 have different gradients? Feel free to define new gradients if needed.

**Comment by ckifer** [comment]
SVG does not support more than one gradient on an element at a time. 

But you should be able to accomplish what you want by using more stops. 

Not at the computer at the moment but I'll try to help in a bit

**Comment by ckifer** [comment, marked as answer]
here you go https://codesandbox.io/p/sandbox/area-chart-filled-by-sign-forked-72qtk2

You'll need to associate colors to your data and re-calculate offset every time a datapoint is added. You'll have a stop per datapoint with the first always being 0 and the last always being 100

![image](https://github.com/user-attachments/assets/24c5a050-a9bf-4835-82a7-af5454894152)

**Reply by cernanb** [reply]
Thank you so much! Just what I was looking for.

**Reply by ckifer** [reply]
No problem, glad it helped!
